pkg/oss/nodeserver.go

```diff
@@ -129,6 +130,11 @@ func validateNodePublishVolumeRequest(req *csi.NodePublishVolumeRequest) error {
 }
 
 func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+	if features.FunctionalMutableFeatureGate.Enabled(features.RundCSIProtocol3) {
+		log.Infof("NodePublishVolume: skip as the rund 3.0 protocol enabled")
+		return &csi.NodePublishVolumeResponse{}, nil
+	}
```

Comment:
What is the logic here? skip the whole NodePublishVolume if a feature gate is enabled?

If this will not become the default in the future, we should still use an env or other config, rather than a feature gate.

Reply:
Yeah it's strange here. We should check the runtimeClass of the pod and only skip mounting for rund, similar to how the feature gate works for disk and nas.

Reply:
This feature gate determines the the kata runtime storage communication protocol with csi which will goes to 3.0 ultimately. why don't we check runtime class in pod first?